core/payment/src/processor.rs

```diff
@@ -585,4 +599,29 @@ impl PaymentProcessor {
         let result = driver_endpoint(&driver).send(msg).await??;
         Ok(result)
     }
+
+    pub async fn shut_down(&mut self, timeout: Duration) {
+        self.in_shutdown = true;
```

Comment:
naming is not consistent: `shutdown` vs `shut_dowm`.. I like the first one more

Reply:
Same issue as 'sendout' vs 'send out'.